File: app/src/main/java/com/github/aakumykov/sync_dir_to_cloud/domain/entities/Abc.java
package com.github.aakumykov.sync_dir_to_cloud.domain.entities;

import java.util.UUID;

public class Abc extends AbcBase {

    public final String id;

    public Abc(AbcBase abcBase) {
        super(abcBase.sourcePath, abcBase.targetPath);
        id = UUID.randomUUID().toString();
    }

    public Abc(String id, String sourcePath, String targetPath) {
        super(sourcePath, targetPath);
        this.id = id;
    }
}

File: app/src/main/java/com/github/aakumykov/sync_dir_to_cloud/domain/entities/qwerty/Abc.kt
package com.github.aakumykov.sync_dir_to_cloud.domain.entities.qwerty

import com.github.aakumykov.sync_dir_to_cloud.domain.entities.AbcBase
import java.util.*

class Abc : AbcBase {

    val id: String

    constructor(abcBase: AbcBase) : super(abcBase.sourcePath, abcBase.targetPath) {
        id = UUID.randomUUID().toString()
    }

    constructor(id: String, sourcePath: String, targetPath: String) : super(
        sourcePath,
        targetPath
    ) {
        this.id = id
    }
}